src/db.rs

```diff
@@ -82,13 +87,12 @@ impl Database {
                 })
                 .expect("Failed to query all tasks")
                 .map(|x| x.expect("Couldn't map over tasks returned by database"))
-                // TODO: remove this if queried by id
-                .filter(|x| !x.finished)
+                .filter(|x| if display_finished { true } else { !x.finished })
```

Comment:
nit: since this is a logic expression and not used to manage control flow, I would personally rewrite this using an logical OR
```suggestion
                .filter(|x| display_finished || !x.finished )
```